### DISCUSSION TITLE
How to fetch data from related tables? (one-to-many)

### DISCUSSION THREAD

**Discussion by edgarasben** [supabase/supabase#3602] [Questions] [answered]
Hey! I am new to Supabase and I am stuck with an issue trying to fetch two related tables (one-to-many).

I have `jobs` and `companies` tables. The `jobs` table has a `company_id` foreign key pointing to `public.companies.id`.

Now I am trying to fetch some jobs with related companies, but this does not seem to work:

```js
import { useState, useEffect } from 'react'
import { supabase } from '../lib/initSupabase'

export default function Index() {

  const [jobs, setJobs] = useState([])

  useEffect(() => {
    fetchJobs()
  }, [])

  async function fetchJobs() {
    let { data: jobsData, error } = await supabase
      .from('jobs')
      .select(`id, title, companies (id, name)`)
    if (error) console.log('error', error)
    else setJobs(jobsData)
  }

  return (
    <ul>
      <li>
        {jobs.map((job) => (
          <div key={job.id}>{job.title}</div>
        ))}
      </li>
    </ul>
  )
}
```

Error I am getting:

```
error
details: [Object, Object]
hint: "By following the 'details' key, disambiguate the request by changing the url to /origin?select=relationship(*) o…"
message: "More than one relationship was found for jobs and companies"
```

Is there something wrong with the way I build a one-to-many relationship or something is wrong with my code?

**Comment by steve-chavez** [comment, marked as answer]
> details: [Object, Object]

Can you print the full objects there? You can use `JSON.stringify()`.

Looks like you could have two foreign keys between jobs and companies, `details` will clarify this.

**Reply by edgarasben** [reply]
details: Array (2)
0 {origin: "public.jobs", relationship: "jobs_company_id_fkey[company_id][id]", cardinality: "m2o", target: "public.companies"}
1 {origin: "public.jobs", relationship: "Jobs_Company_fkey[company_id][id]", cardinality: "m2o", target: "public.companies"}

I see now. There might be a leftover relationship. Maybe when I renamed tables. Is there a way to delete it?

**Reply by steve-chavez** [reply]
>  Maybe when I renamed tables. Is there a way to delete it?

You can do it with SQL:

```sql
ALTER TABLE public.jobs DROP CONSTRAINT "Jobs_Company_fkey";
```

**Reply by edgarasben** [reply]
Awesome! Do you also think it's a bug? Maybe renaming tables should also rename the constraints?

**Reply by steve-chavez** [reply]
> Do you also think it's a bug?

Yeah, I think that could be improved. Maybe the issue is that a rename produces another constraint.

Filing an issue with a reproducible test case would be great :pray:

**Reply by edgarasben** [reply]
> > Do you also think it's a bug?
> 
> 
> 
> Yeah, I think that could be improved. Maybe the issue is that a rename produces another constraint.
> 
> 
> 
> Filing an issue with a reproducible test case would be great :pray: 

Good idea, will try tomorrow! What's the sql query to see all current constraints on a table btw?

**Reply by steve-chavez** [reply]
> What's the sql query to see all current constraints on a table btw?

Check here! https://database.dev/list-all-constraint
